### PR TITLE
feat: add numaflow controller definition for v1.3.0

### PR DESCRIPTION
**Pull request by dpadhiar**
### Modifications

Adding the controller definition for `v1.3.0` which is the most recent Numaflow version that includes monovertices.

### Verification

Ran `make start` to update local cm and `make test` to verify tests still work using this definition.

Created sample rollouts to see that the resources were created and functioned properly using this new definition.

### CODE REVIEW

File: tests/manifests/default/controller_definitions.yaml [hunk 1]
```diff
@@ -248,141 +286,115 @@ controllerDefinitions:
...
+            app.kubernetes.io/component: controller-manager
+            app.kubernetes.io/name: controller-manager
+            app.kubernetes.io/part-of: numaflow
+        spec:
+          containers:
+          - args:
+            - controller
+            env:
+            - name: NUMAFLOW_IMAGE
+              value: quay.io/numaproj/numaflow:1.3.0-rc1
```
Comment (vigith) [inline tests/manifests/default/controller_definitions.yaml:321]:
```suggestion
              value: quay.io/numaproj/numaflow:1.3.0
```

we have released 1.3.0

Comment (juliev0) [reply tests/manifests/default/controller_definitions.yaml:321]: Do you want to replace that here @dpadhiar ?

**Comment by juliev0** [comment]
I think the way to verify this is to create your sample NumaflowControllerRollout and then the other Rollouts, and make sure everything gets created successfully. (and no weird errors in Numaflow log)

File: tests/manifests/default/controller_definitions.yaml (hunk 1 above)
Comment (dpadhiar) [reply tests/manifests/default/controller_definitions.yaml]: Updated now to use image `v1.3.0`

**Review by xdevxy — APPROVED** [review]

**Review by juliev0 — APPROVED** [review]

**Pull request merged**